js/index.js

```diff
@@ -141,52 +140,18 @@ Raven.context(async () => {
     document.getElementById("app").style.visibility = "hidden";
     return;
   }
-  const __extraWindows = [];
-  let __initialWindowLayout = null;
 
+  let butterchurnConfig = null;
   if (isButterchurnSupported()) {
     const startWithMilkdropHidden =
       document.body.clientWidth < MIN_MILKDROP_WIDTH ||
       (!library && skinUrl != null) ||
       screenshot;
 
-    __extraWindows.push({
-      id: "milkdrop",
-      title: "Milkdrop",
-      isVisualizer: true,
-      Component: MilkdropWindow,
-      open: !library && !startWithMilkdropHidden
-    });
-
-    if (startWithMilkdropHidden) {
-      __initialWindowLayout = {
-        [WINDOWS.MAIN]: { position: { x: 0, y: 0 } },
-        [WINDOWS.EQUALIZER]: { position: { x: 0, y: 116 } },
-        [WINDOWS.PLAYLIST]: { position: { x: 0, y: 232 }, size: [0, 0] },
-        milkdrop: { position: { x: 0, y: 348 }, size: [0, 0] }
-      };
-      if (library) {
-        __initialWindowLayout[WINDOWS.MEDIA_LIBRARY] = {
-          position: { x: 0, y: 348 },
-          size: [0, 0]
-        };
-      }
-    } else {
-      __initialWindowLayout = {
-        [WINDOWS.MAIN]: { position: { x: 0, y: 0 } },
-        [WINDOWS.EQUALIZER]: { position: { x: 0, y: 116 } },
-        [WINDOWS.PLAYLIST]: { position: { x: 0, y: 232 }, size: [0, 4] },
-        milkdrop: { position: { x: 275, y: 0 }, size: [7, 12] }
-      };
-      if (library) {
-        __initialWindowLayout[WINDOWS.MEDIA_LIBRARY] = {
-          position: { x: 275, y: 0 },
-          size: [7, 12]
-        };
-      }
-    }
-
-    document.getElementById("butterchurn-share").style.display = "flex";
+    butterchurnConfig = {
+      open: !library && !startWithMilkdropHidden,
+      options: butterchurnOptions
+    };
```

Comment:
Having both "options" and "config" seems like it's going to be a source of confusion. Let's just name the Webamp config property name: `__butterchurnOptions`. On this line we can just define a `butterchurnOpen` variable. Then, when we can combine them into a single flat object (where `open` is a direct sibling of the other options) down below where we actually pass it to Webamp. If you want, you could rename the imported `butterchurnOptions` to `partialButterchurnOptions`.